osu.Server.Queues.ScoreStatisticsProcessor/Processors/MedalAwarders/ModIntroductionMedalAwarder.cs

```diff
@@ -0,0 +1,69 @@
+﻿using JetBrains.Annotations;
+using MySqlConnector;
+using osu.Game.Online.API.Requests.Responses;
+using osu.Game.Rulesets;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Osu.Mods;
+using osu.Server.Queues.ScoreStatisticsProcessor.Helpers;
+using osu.Server.Queues.ScoreStatisticsProcessor.Models;
+using System.Collections.Generic;
+
+namespace osu.Server.Queues.ScoreStatisticsProcessor.Processors.MedalAwarders
+{
+    [UsedImplicitly]
+    public class ModIntroductionMedalAwarder : IMedalAwarder
+    {
+        public bool RunOnFailedScores => false;
+
+        public IEnumerable<Medal> Check(SoloScoreInfo score, UserStats userStats, IEnumerable<Medal> medals, MySqlConnection conn, MySqlTransaction transaction)
+        {
+            // Ensure the mod is the only one selected
+            if (score.Mods.Length != 1)
+                yield break;
```

Comment:
This is the legacy web implementation yes but I am not sure that it should continue to work this way. The primary counterexample is classic mod: I'm not sure that a classic + perfect play should not award the requisite medal.

I'm not sure there are any other exceptions one might want to allow other than classic but food for thought.

Reply:
Classic mod is currently disallowed for ranking (other than for legacy score), which is how I implemented this.

I tried to do this in a way where this would be straight-forward to expand with other mods when/if desired.

Reply:
Ranking is not medal granting though. The two are independent. Classic mod plays will submit and show on leaderboards but will not give pp.

Reply:
Okay, changed this to also apply on non-legacy classic scores.

Reply:
imo the correct mods to exclude here would be Classic as well as all "System" type mods. the point of these medals, how i understand them, is that the player tried the mod by itself to see how it works -- but the player has no direct control over system mods.

Reply:
@cl8n should be addressed with 2a8da5deaca77490c653af214a0f87e3f20db25f (with test coverage at 23774c1651c873ef249577459aeb8c445126f092